pomice/pool.py

```diff
@@ -545,11 +545,19 @@ async def build_track(self, identifier: str, ctx: Optional[commands.Context] = N
             path="decodetrack",
             query=f"encodedTrack={quote(identifier)}",
         )
+
+        if self._version.major >= 4:
+            track_info = data["info"]
+            track_type = data["info"]["sourceName"]
+        else:
+            track_info = data
+            track_type = data["sourceName"]
```

Comment:
i don't see why this is necessary?

wouldn't this always be equivalent to `track_info["sourceName"]`?

Reply:
`track_info["sourceName"]` results in a `KeyError` because the new format puts the `sourceName` inside of the `info` object.
So on v3 instances, it's still in the root of the dict, but on v4 instances, it's underneath the `info` dict.

Reply:
Should be resolved now :)